source/Halibut/Transport/SecureListener.cs

```diff
@@ -118,49 +115,74 @@ public void Disconnect(string thumbprint)
 
         void Accept()
         {
-            var numberOfFailedAttemptsInRow = 0;
-
-            while (!cts.IsCancellationRequested)
-            {
-                try
+            // See: https://github.com/OctopusDeploy/Issues/issues/6035
+            // See: https://github.com/dotnet/corefx/issues/26034
+            var tokenCancellationListener = IsWindows()
+                ? cts.Token.Register(listener.Stop)
+                : (IDisposable) new NoopDisposable();
```

Comment:
You can just use `null` here, `using(null)` totally works. I'd also move the declaration closer to the using statement, or even within it `using(IsWindows() ? cts.Token.Register(listener.Stop) : null)`

Reply:
Nice! I'd never tried `using(null)` before. That is good to know!